src/main/java/sk/uniba/fmph/dcs/game_board/ToolMakerHutFields.java

```diff
@@ -0,0 +1,121 @@
+package sk.uniba.fmph.dcs.game_board;
+
+import org.json.JSONObject;
+import sk.uniba.fmph.dcs.stone_age.Effect;
+import sk.uniba.fmph.dcs.stone_age.InterfaceGetState;
+import sk.uniba.fmph.dcs.stone_age.PlayerOrder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class ToolMakerHutFields implements InterfaceGetState {
+
+    private List<PlayerOrder> toolMakerFigures;
+    private List<PlayerOrder> hutFigures;
+    private List<PlayerOrder> fieldsFigures;
+    private int restriction;
+    private int locationsOccupied;
+
+    public ToolMakerHutFields() {
+        toolMakerFigures = new ArrayList<>();
+        hutFigures = new ArrayList<>();
+        fieldsFigures = new ArrayList<>();
+        locationsOccupied = 0;
+    }
+
+    public boolean placeOnToolMaker(Player player) {
+        if (!canPlaceOnToolMaker(player)) return false;
+        locationsOccupied++;
+        toolMakerFigures.add(player.getPlayerOrder());
+        player.getPlayerBoard().takeFigures(1);
+        return true;
+    }
+
+    public boolean actionToolMaker(Player player) {
+        if (!toolMakerFigures.contains(player.getPlayerOrder())) return false;
+        player.getPlayerBoard().takeFigures(-1);
+        player.getPlayerBoard().giveEffect(new Effect[] {Effect.TOOL});
+        toolMakerFigures.clear();
```

Comment:
to by to malo asi aj urobit locationsOccupied--, ale nic to nemeni na fungovani, len moze dat zly state string